Streaming HF Dataset - No chunks found issue

## 🐛 Bug

I am trying to Stream a HuggingFace database and have followed the steps in the README but I keep having a No chunks found error. This happens regardless of whether I explicitly try to index the dataset first or not.

### To Reproduce

Steps to reproduce the behavior...

Approach 1:
```
hf_dataset_uri = "hf://datasets/datamol-io/safe-gpt/"
dataset = ld.StreamingDataset(hf_dataset_uri)
```
Approach 2:
```
ld.index_hf_dataset(hf_dataset_uri, index_path)
dataset = ld.StreamingDataset(hf_dataset_uri)
```
Error:
```
AssertionError: No chunks found in the `Dir(path='/u/shreyans/.lightning/chunks/ba4d7049ae87cc2738503ad50a639355/1747772774.5207303', url='hf://datasets/Qwen/PolyMath/')/index.json` file
```
and index.json has the following:
```
{"chunks": [], "config": {"chunk_bytes": null, "chunk_size": null, "compression": null, "data_format": [], "data_spec": null, "encryption": null, "item_loader": "ParquetLoader"}, "updated_at": "1747753650.7237446"}
```

I am confused as to whether I need to explicitly define the chunks somehow? I thought this would happen automatically as part of the indexing.

### Expected behavior

Expected an index to be created with defined chunks and then to be able to access items from the dataset.

<details>
  <summary>Environment detail</summary>

- PyTorch Version (e.g., 1.0): 2.2.0+cu121
- OS (e.g., Linux): Linux
- How you installed PyTorch (`conda`, `pip`, source): conda
- Python version: 3.11.12

</details>

--- a/src/litdata/utilities/parquet.py
+++ b/src/litdata/utilities/parquet.py
@@ -39,7 +39,7 @@ class ParquetDir(ABC):
             raise RuntimeError(
                 f"No Parquet files were found at '{self.dir.url or self.dir.path}'. "
                 "Please verify that the provided path is correct and that it contains at least one .parquet file. "
-                "If your files are located in a subdirectory, please specify the correct path."
+                "If the files are located in a subdirectory, please specify the correct path."
             )
 
         with ThreadPoolExecutor(max_workers=self.num_workers) as executor:

--- a/src/litdata/utilities/parquet.py
+++ b/src/litdata/utilities/parquet.py
@@ -35,6 +35,13 @@ class ParquetDir(ABC):
         Yields:
             Generator[Tuple[str, int], None, None]: A generator yielding tuples of file name, file path, and order.
         """
+        if not self.files:
+            raise RuntimeError(
+                f"No Parquet files were found at '{self.dir.url or self.dir.path}'. "
+                "Please verify that the provided path is correct and that it contains at least one .parquet file. "
+                "If your files are located in a subdirectory, please specify the correct path."
+            )
+
         with ThreadPoolExecutor(max_workers=self.num_workers) as executor:
             futures = {executor.submit(self.task, _file): (order, _file) for order, _file in enumerate(self.files)}
             for future in futures:

--- a/tests/streaming/test_parquet.py
+++ b/tests/streaming/test_parquet.py
@@ -203,3 +203,24 @@ def test_cache_dir_option(monkeypatch, huggingface_hub_fs_mock, default):
             pass
         # check chunk cache dir was filled
         assert len([f for f in os.listdir(ds.input_dir.path) if f.endswith(".parquet")]) == 5  # 5 chunks
+
+
+@pytest.mark.parametrize(
+    ("pq_url"),
+    [
+        "s3://some_bucket/some_path",
+        "gs://some_bucket/some_path",
+        "hf://datasets/some_org/some_repo/some_path",
+    ],
+)
+@patch("litdata.utilities.parquet._HF_HUB_AVAILABLE", True)
+@patch("litdata.streaming.downloader._HF_HUB_AVAILABLE", True)
+@patch("litdata.utilities.parquet._FSSPEC_AVAILABLE", True)
+def test_no_parquet_files(pq_url, tmpdir, huggingface_hub_fs_mock, fsspec_pq_mock):
+    ls_mock = Mock()
+    ls_mock.ls = Mock(side_effect=lambda *args, **kwargs: [])
+    huggingface_hub_fs_mock.HfFileSystem = Mock(return_value=ls_mock)
+    fsspec_pq_mock.filesystem = Mock(return_value=ls_mock)
+
+    with pytest.raises(RuntimeError, match="No Parquet files were found"):
+        index_parquet_dataset(pq_url, cache_dir=tmpdir)

